includes/abstracts/abstract-wc-product.php

```diff
@@ -1393,17 +1404,38 @@ public function save() {
 		return $this->get_id();
 	}
 
+	/**
+	 * Do any extra processing needed before the actual product save
+	 * (but after triggering the 'woocommerce_before_..._object_save' action)
+	 *
+	 * @return mixed A state value that will be passed to after_data_store_save_or_update.
+	 */
+	protected function before_data_store_save_or_update() {
```

Comment:
I am assuming this is intentionally left blank for future changes?

Reply:
This is intentionally blank indeed, to be filled by derived classes if needed (as it happens in `class-wc-product-variable.php`).